### DISCUSSION TITLE
Preserve info about previous state after executing wfc

### DISCUSSION THREAD

**Discussion by sjrc6** [mxgmn/MarkovJunior#51] [Q&A]
Hello! Maybe this is already possible but it would be very helpful to be able to keep certain colors from before wfc exectues to after it finishes, 

I'm trying to run wfc on just a specific area and then fill in around that area when it's done but running wfc makes it hard to preserve any information about the previous state since it seems like overrides any colors not in the sample/tileset.

for example, I want to run wfc on the white parts of this path, then afterwards fill in the black parts with a different color so I can isolate the original path again. 
![image](https://user-images.githubusercontent.com/22122579/182948431-5aab55a3-a7c2-4137-a965-36cdfb85a641.png)

but when wfc finishes I don't have any info about which parts used to be white or black so it's impossible to fill in the gaps with the `path` node while preserving the shape of the previous white path. 
![image](https://user-images.githubusercontent.com/22122579/182948682-495c80d4-fb1e-43af-aefd-0fd561749464.png)

I couldn't find this in any of the examples so I'm not sure if it's currently possible.

**Comment by mxgmn** [comment]
Hi! Indeed, it's not possible right now. In tiled WFC, you can circumvent this issue by making a second empty tile with the same adjacencies as the first one, and map original black squares to second-kind empty only. I don't see how to do something similar with overlapping WFC.

I'm actually working right now on extending the language with a stack that can keep intermediate grids and other values. But these features won't be made public soon.

**Reply by sjrc6** [reply]
That sounds very exciting! Thanks for the quick response. I found the ConvChain node is good enough for now and much more controllable. I had actually thought about that workaround but the WFC tilesets take a long time to make so I was hoping there was another way.

**Reply by mxgmn** [reply]
You can also do this trick with convolution nodes. Run `ConstrainedCaves` for an example.
